documentation/internal/Assignment-Process.md

```diff
@@ -0,0 +1,57 @@
+# Assignment Process
+
+The logic of assigning a review spans between both web app and server, front end is only concerned with queries and mutations to review_assignment and review_question_assignment tables, whereas back end does most of the grunt work (through actions and triggers) and generates records in those tables based on user permissions and states of application and it's review.
+
+## Schema
+
+Seems like a good starting (anchor) point for describing this functionality, let's have look:
+
+![Review Assignment](images/Review-Assignment.png)
+
+First of all, some bulletpoint (and rules) to help clarify the diagram
+
+- One review assignment should exists per reviewer/application/stage/level
+- Review assignment can have the following statuses
+  - `Available` -> Can be assigned by assigner
+  - `Self-assigned by another` -> Was available for self assignment (but self assigned by someone else)
+  - `Assigned` -> At least on review_question_assignment was linked to review_assignment
+  - `Available for self-assignment` -> Can be self assigned
+- Review assignments can be restricted by section (template_section_restriction, which is an array of section codes)
+- Review assignments of status `Available or Assigned` can be assigned by the assigner linked through the join table
+- `Assigned` status is used to indicate to the reviewer that they can start the review (if review hasn't been started already)
+- Review assignment should only be editable by a reviewer if it's `Available for self-assignment`, otherwise by linked `Assigner`. Review assignment should not be editable once stage of review assignmetn is different to application stage
```

Comment:
```suggestion
- Review assignment should only be editable by a reviewer if it's `Available for self-assignment`, otherwise by linked `Assigner`. Review assignment should not be editable once stage of review assignment is different to the current application stage.
```